### PR TITLE
Fix to bounds provider code which inadvertently returns NaN during init.

### PR DESCRIPTION
**Pull request by gerring**
https://github.com/eclipse/richbeans/issues/134

### CODE REVIEW

File: org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java [hunk 1]
```diff
@@ -1112,11 +1116,11 @@ public void setFieldOveride(String fieldOveride) {
 		this.fieldOveride = fieldOveride;
 	}
 	
-	public IBeanController getBeanService() {
+	public IBeanController<?> getBeanService() {
```
Comment (gerring) [inline org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java]: CRITICAL Remove usage of generic wildcard type. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1452)

**Comment by gerring** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java [hunk 2]
```diff
@@ -98,7 +98,7 @@
 	// FIXME Widgets should not have a reference to the controller.
 	// This breaks the model/view/controller design because the controller
 	// is referenced by the view to look up field values.
-	private IBeanController beanService;
+	private IBeanController<Object> beanService;
```
Comment (jamesmudd) [inline org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java]: There is no code in GDA that uses this IBeanController outside this class so yes it should be removed. Will improve encapsulation and make it a bit more MVCish.

File: org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java [hunk 3]
```diff
@@ -781,8 +781,10 @@ public void setDecimalPlaces(int decimalPlaces) {
 	 */
 	@SuppressWarnings("unchecked")
 	public double getMaximum() {
-		if (maxProvider != null)
-			return maxProvider.getBoundValue();
+		if (maxProvider != null) {
+			double dbl = maxProvider.getBoundValue();
+			if (!Double.isNaN(dbl)) return dbl;
```
Comment (jamesmudd) [inline org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java]: Should this throw if db1 is null? Presumably that is the case where a maxProvider is set but it has given a null bound?

**Review by jamesmudd — APPROVED** [review]
Think this looks better but won't fix my case I never set a BoundsProvider

File: org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java (hunk 3 above)
Comment (gerring) [inline org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java]: dbl is a primitive which cannot be null. There is an event which comes through there the value is NaN during linking the UI. This event should not be sent but was a more risky change :) 

Thanks for checking about IBeanController, I think I will add its removal to this PR.

**Pull request merged**